spec/rate_limit_control_spec.rb

```diff
@@ -3,7 +3,47 @@
     expect(RateLimitControl::VERSION).not_to be nil
   end
 
-  it "does something useful" do
-    expect(false).to eq(true)
+  after { Redis.current.flushdb }
+
+  let(:action_config) do
+    {
+      action: 'rate_limit_example',
+      id: 'foobar',
+      allowed_requests: 3,
+      storage: Redis.new,
+      timeout: 10
+    }
+  end
+
+  let(:action) { "Any action you would like to execute" }
+
+  context "when the actions do not exceed the total allowed" do
+    before { expect_any_instance_of(described_class).not_to receive(:set_action_as_blocked) }
+
+    it "runs the requested actions" do
+      3.times { described_class.new(action_config) { action } }
+    end
+  end
+
+  context "when the actions exceed the total allowed" do
+    before { expect_any_instance_of(described_class).to receive(:set_action_as_blocked).once }
+
+    it 'sets the action as blocked' do
+      4.times { described_class.new(action_config) { action } }
+    end
+  end
+
+  context "when an action stack is full" do
+    before do
+      expect_any_instance_of(described_class).not_to receive(:set_action_as_blocked)
+
+      3.times { described_class.new(action_config) { action } }
+    end
+
+    it "does not block other actions" do
+      action_config[:action] = 'rate_limit_example_2'
```

Comment:
I think you could create a new let here with this different config or create a let just for the action name.

Like:
```
  let(:another_action_config) do
    {
      action: 'rate_limit_example_2',
      id: 'foobar',
      allowed_requests: 3,
      storage: Redis.new,
      timeout: 10
    }
  end
```

Or

```
  let(:action_name) { "rate_limit_example" }
  let(:another_action_config) do
    {
      action: action_name,
      id: 'foobar',
      allowed_requests: 3,
      storage: Redis.new,
      timeout: 10
    }
  end
```

And in this `it` you can create another `action_name`.

Reply:
I like the idea!